spec/lib/services/dialog_import_service_spec.rb

```diff
@@ -144,10 +142,28 @@
   end
 
   describe "#import_all_service_dialogs_from_yaml_file" do
-    include_context "DialogImportService dialog setup"
+    let(:dialog_fields) do
+      [{"name" => "dialog_field1"},
+       {"name" => "dialog_field2", "dialog_field_responders" => ["dialog_field"] }]
+    end
+
+    let(:dialog_groups) do
+      [{"label" => "New Box", "dialog_fields" => dialog_fields}]
+    end
+
+    let(:dialog_tabs) do
+      [{"label" => "New Tab", "dialog_groups" => dialog_groups}]
+    end
+
+    let(:dialog) do
+      [{"label" => "Test2", "dialog_tabs" => dialog_tabs, "description" => "potato"}]
+    end
 
     before do
-      allow(YAML).to receive(:load_file).with("filename").and_return(dialogs)
+      built_dialog_field = DialogField.create(:name => "dialog_field")
```

Comment:
It sounds like the name sequencing belongs in the factory, can you add that logic there?